packages/toolpad-app/src/server/logs/logger.ts

```diff
@@ -21,17 +21,42 @@ if (config.ecsNodeUrl) {
 
 const logger = pino(
   {
-    enabled: config.serverLogsEnabled,
     level: process.env.LOG_LEVEL || 'info',
     redact: { paths: [] },
     serializers: {
+      err: errSerializer,
+      error: errSerializer,
       req: reqSerializer,
       res: resSerializer,
-      resErr: resErrSerializer,
     },
     ...(config.ecsNodeUrl ? ecsFormat() : {}),
   },
   transport,
 );
 
-export default logger;
+interface ReqResLogPayload {
+  key: 'apiReqRes';
+  req: NextApiRequest;
+  res: NextApiResponse;
+}
+
+interface RpcReqResLogPayload {
+  key: 'rpc';
+  type: 'query' | 'mutation';
+  name: string;
+  error: Error | null;
+}
+
+type LogPayload = ReqResLogPayload | RpcReqResLogPayload;
+
+function logMethod(method: 'info' | 'trace' | 'error' | 'warn' | 'fatal') {
```

Comment:
nice, this works better than a function just for `logInfo`

Reply:
Yes, I'm thinking roughly:

* `fatal`: Unrecoverable errors, the node process should crash (or has crashed and this is the report).
* `error`: Unexpected, but we can degrade in a safe way. The user experience is impacted, so this must be investigated
* `warn`: Unexpected, but we can degrade in a safe way. The user experience isn't really impacted, investigate when lots of these happen.
* `info`: informational, we want to see these appear in a log aggregator for metrics


We play with logelevel on different use-cases
* production: `info`
* local: `warn`, info will most likely lead to too many logs

Reply:
Sounds great, one of the alerts I set up in Sentry was for frequent warnings too.